### PR TITLE
Add fully automated restore to REQUESTRESTORE

### PR DESCRIPTION
**Pull request by schlomo**
##### Pull Request Details:

* Type: **Enhancement**

* Impact: **Low**

* How was this pull request tested?

Manually on Ubuntu 22.04

* Brief description of the changes in this pull request:

Add an automated restore feature to `REQUESTRESTORE`, similar to how `NFS4SERVER` works.

Example:

```
Confirm the recreated disk layout or go back one step
1) Confirm recreated disk layout and continue 'rear recover'
2) Go back one step to redo disk layout recreation
3) Use Relax-and-Recover shell and return back to here
4) Abort 'rear recover'
(default '1' timeout 300 seconds)

User confirmed recreated disk layout
Now you must manually restore your backup.
Ensure the backup data gets restored into /mnt/local (instead of '/')
because the disks and filesystems of the target system are mounted there.

Waiting for /mnt/local/ok file to signal when the restore is completed and recovery can proceed.
Waiting since 02:05, used storage space: 4.3G
Restored 4.4G in 02:10 minutes.
Created SELinux /mnt/local/.autorelabel file : after reboot SELinux will relabel all files
Recreating directories (with permissions) from /var/lib/rear/recovery/directories_permissions_owner_group
```

with a config like this:

```sh
REQUESTRESTORE_FINISHED_FILE=ok
REQUESTRESTORE_ABORT_FILE=bad
```

Without those two variables the behaviour is like before, with a prompt and dropping the user into `rear_shell`.

This will also help with #2988 for this backup method.

I used the following command on another computer to perform the remote restore of a backup created via `BACKUP=NETFS`

```sh
ssh root@192.168.11.52 "tar -C /mnt/local -x -v -z && echo >/mnt/local/ok" </srv/scratch/rear-u2204/backup.tar.gz
```

@codefritzel what do you think?

@rear/contributors please have a look, I'd like to merge this soon

### CODE REVIEW

**Comment by codefritzel** [comment]
```
Waiting since 02:05, used storage space: 4.3G
Restored 4.4G in 02:10 minutes.
```
The output is a very good idea and also adopted for `NFS4SERVER`.

File: usr/share/rear/conf/default.conf [hunk 1]
```diff
@@ -2847,6 +2847,15 @@ because the disks and filesystems of the target system are mounted there.
 # This command is shown to the user and added to the predefined bash history
 # in the ReaR recovery system to make the manual backup restore easier:
 REQUESTRESTORE_COMMAND=
+#
+# Set the finished and abort files to enable non-interactive REQUESTRESTORE mode.
+# ReaR will wait for the finished file to appear and continue with the recovery,
+# or abort the recovery if the abort file appears. The content of the abort file is
+# shown and logged as abort reason.
+# Instead of the manual prompt ReaR will simply show an info message and the disk
+# usage of the target filesystems.
```
Comment (codefritzel) [inline usr/share/rear/conf/default.conf:2856]: I would add here that if these variables are empty, `REQUESTRESTORE` works as before.

File: usr/share/rear/lib/filesystems-functions.sh [hunk 2]
```diff
@@ -239,3 +239,8 @@ function xfs_parse
   # Output xfs options for further use
   echo "$xfs_opts"
 }
+
+# return the total used disk space of the target file systems
+function total_target_fs_used_disk_space() {
+    df --total --local -h --exclude-type=tmpfs --exclude-type=devtmpfs $(mount | grep "$TARGET_FS_ROOT" | cut -f 1 -d " ") | awk 'END{print $3}'
+}
```
Comment (codefritzel) [inline usr/share/rear/lib/filesystems-functions.sh:246]: a line-break is missing here

**Review by codefritzel — APPROVED** [review]

File: usr/share/rear/restore/REQUESTRESTORE/default/200_prompt_user_to_start_restore.sh [hunk 3]
```diff
@@ -5,21 +5,44 @@
...
+        Error "Couldn't delete restore finished file $check_file or abort file $abort_file"
+    LogPrint "Waiting for $check_file file to signal when the restore is completed and recovery can proceed."
+    (( start = SECONDS ))
+    until test -f "$check_file" ; do
+        test -f "$abort_file" && \
+            Error "Restore aborted by abort file $abort_file. Reason given:$LF$(< "$abort_file"))"
+        (( duration = SECONDS - start ))
+        printf -v runtime "%02d:%02d" $(( duration/60 )) $(( duration % 60 ))
+        ProgressInfo "Waiting for $runtime minutes, total used storage space: $(total_target_fs_used_disk_space)"
+        sleep 5
```
Comment (jsmeix) [inline usr/share/rear/restore/REQUESTRESTORE/default/200_prompt_user_to_start_restore.sh:22]: We have PROGRESS_WAIT_SECONDS for this

File: usr/share/rear/restore/NFS4SERVER/default/400_restore_with_nfs_server.sh [hunk 4]
```diff
@@ -10,16 +10,16 @@ rm -f "$check_file" "$abort_file" || Error "Couldn't delete restore finished fil
...
         Error "Restore aborted by abort file $abort_file. Reason given:$LF$(< "$abort_file"))"
     fi
-    used_space=$(
-        df --total --local -h --exclude-type=tmpfs --exclude-type=devtmpfs | awk 'END{print $3}'
-        ) && \
-        ProgressInfo "      Used storage space: $used_space"
+    (( duration = SECONDS - start ))
+    printf -v runtime "%02d:%02d" $(( duration/60 )) $(( duration % 60 ))
+    ProgressInfo "Waiting for $runtime minutes, total used storage space: $(total_target_fs_used_disk_space)"
     sleep 5
```
Comment (jsmeix) [inline usr/share/rear/restore/NFS4SERVER/default/400_restore_with_nfs_server.sh:23]: We have PROGRESS_WAIT_SECONDS for this

**Review by jsmeix — APPROVED** [review]
I did not test it myself - I just trust you.

File: usr/share/rear/lib/filesystems-functions.sh (hunk 2 above)
Comment (jsmeix) [reply usr/share/rear/lib/filesystems-functions.sh:247]: the newline at the end of the file is still missing here, cf.
https://github.com/rear/rear/pull/2973#issuecomment-1521648590

**Comment by schlomo** [comment]
I fixed everything, please have a look

**Review by jsmeix — APPROVED** [review]

**Pull request merged**